File: service-common/src/main/java/me/reidj/bridgebuilders/tops/PlayerTopEntry.java
package me.reidj.bridgebuilders.tops;

import lombok.Getter;
import lombok.Setter;
import me.reidj.bridgebuilders.user.Stat;

@Setter
@Getter
public class PlayerTopEntry<V> extends TopEntry<Stat, V> {

    private String userName;
    private String displayName;

    public PlayerTopEntry(Stat stat, V value) {
        super(stat, value);
    }

}

File: commons/src/main/kotlin/me/reidj/bridgebuilders/top/PlayerTopEntry.kt
package me.reidj.bridgebuilders.top

import me.reidj.bridgebuilders.user.Stat

data class PlayerTopEntry<V>(override var key: Stat, override var value: V): TopEntry<Stat, V>(key, value) {

    lateinit var userName: String
    lateinit var displayName: String
}
